### PR TITLE
perf(ci): only publish artifacts on git version tags of main 

### LINKED ISSUE #2360: perf(ci): only publish artifacts on git version tags of main
### Description
As a maintainer and/or contributor I want to have fast CI cycles so that I'm more productive and the community as a whole is growing faster because the contributions can be accepted at a rapid pace instead of the CI bogging everybody down.

Right now we have a lot of publishing tasks that take up resources (mostly) needlessly because they auto-publish artifacts such as container images and JVM API clients on every merged commit.
This has the negative effect of slowing the CI down overall and also the other negative effect that we are risking hitting limitations on the free tiers that we are using (limits where the number of versions must stay below a certain number for npm packages or container images for example).

One example of this auto-publishing is the kotlin [1] packages and all the other `**-publish.yaml` files that we have for the container images.

[1] `.github/workflows/ghpkg-all-kotlin-api-clients-publish.yaml`

### Acceptance Criteria
1. Go through every single .yaml file and produce a list of them which are currently doing auto-publishing on every commit to `main`
2. Refactor the list of yaml's obtained in step 1 so that artifact publishing only happens on git tags.
3. Make sure that the refactor happens in a way that the container image builds are still being tested as part of every PR.

### PR DESCRIPTION
**Pull request by raynatopedrajeta**
### **Commit** to be reviewed
---
perf(ci): only publish artifacts on git version tags of main 
```
Primary Changes 
----------------
1. Remove auto publishing of container images in all **publish.yaml upon push and maintain automation runs during pull requests
```
Fixes #2360

**Pull Request Requirements**
- [ ] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [ ] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [ ] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [ ] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [ ] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]
The publishing of artifacts shall happen only upon the creation of a tag. Requesting changes for the same

File: .github/workflows/all-nodejs-packages-publish.yaml
Comment (petermetz) [inline .github/workflows/all-nodejs-packages-publish.yaml:11]: @raynatopedrajeta Wouldn't adding this make it so that this workflow gets executed for every pull request instead of just the tags starting with the character `v`? 
It's been a while since I've dealt with these configurations so I'm not sure, but it looks like it would be doing the opposite of the task, e.g. instead of reducing the resource usage it would increase it.

**Review by petermetz — COMMENTED** [review]
^^ Please see https://www.linkedin.com/posts/hyperledger-project_hyperledger-cacti-interop-activity-7196887184212148225-FyE0

**Comment by raynatopedrajeta** [comment]
Hello @petermetz ,

Review point has been addressed. Re-requested for review.

FYI @jagpreetsinghsasan 

Thanks!
Ren

**Comment by jagpreetsinghsasan** [comment]
Congratulations @raynatopedrajeta for your first PR to Hyperledger Cacti ! 
Keep up the good work 🥳

**Comment by raynatopedrajeta** [comment]
> Congratulations @raynatopedrajeta for your first PR to Hyperledger Cacti ! Keep up the good work 🥳

Thanks a lot @jagpreetsinghsasan ! Looking forward for more learnings with the team!

**Review by petermetz — CHANGES_REQUESTED** [review]
@raynatopedrajeta Please resolve the merge conflicts, other than that it should be good to go from my side.

**Comment by petermetz** [comment]
@jagpreetsinghsasan Were your concerns addressed?

**Comment by jagpreetsinghsasan** [comment]
> @jagpreetsinghsasan Were your concerns addressed?

Yes @petermetz , the diff looks fine now

**Comment by petermetz** [comment]
@jagpreetsinghsasan OK, thank you.

@raynatopedrajeta Please rebase and resolve the conflicts and then re-request review from both me and @jagpreetsinghsasan and we'll get this merged ASAP.

**Comment by raynatopedrajeta** [comment]
Hello @petermetz  and @jagpreetsinghsasan ,
 Review changes has been addressed.

Thanks!

**Comment by jagpreetsinghsasan** [comment]
LGTM. Thanks for a quick rebase, @raynatopedrajeta

**Review by petermetz — APPROVED** [review]

**Review by petermetz — CHANGES_REQUESTED** [review]
@raynatopedrajeta Oops, one more nit: Please fix the ActionLint check that's been failing:

image

**Comment by raynatopedrajeta** [comment]
Hello @petermetz ,

I have pushed another fix for the action lint. However I can't run the workflows yet since it needs maintainer's approval. 

Please let me know if there will be more changes.

Thanks!

Raynato

**Comment by jagpreetsinghsasan** [comment]
> Hello @petermetz ,
> 
> I have pushed another fix for the action lint. However I can't run the workflows yet since it needs maintainer's approval.
> 
> Please let me know if there will be more changes.
> 
> Thanks!
> 
> Raynato

Done @raynatopedrajeta  !!

**Review by petermetz — APPROVED** [review]
@raynatopedrajeta LGTM, thank you for the updates!

**Review by outSH — APPROVED** [review]

**Pull request merged**
